src/g_game.c

```diff
@@ -773,7 +773,7 @@ void G_BuildTiccmd(ticcmd_t* cmd)
       // killough 10/98: make SG/SSG and Fist/Chainsaw
       // weapon toggles optional
       
-      if (!demo_compatibility && doom_weapon_toggles)
+      if (!demo_compatibility && doom_weapon_toggles && !next_weapon)
```

Comment:
Okay, so cycling next/prev goes through all weapons for complevel 9 or later, regardless of `doom_weapon_toggles`. That's fine and makes sense.

At some point in the past, Chocolate Doom's method of cycling next/prev became the standard for vanilla behavior, where cycling next only selects the highest slot weapons, and cycling prev goes through all weapons (with chainsaw/fist berserk restriction). I understand why. It's closest to vanilla behavior, which never had next/prev weapon, and it was required to press "1" or "3" twice to get the next lower weapon.

So my question is, do we HAVE to follow Chocolate Doom's method for complevel 2? Going from Pistol to Shotgun directly (skipping SSG) is an unfair advantage, right? Well how about we only enforce that with strict mode? Otherwise, allow full weapon selection for any complevel by default. And allow `doom_weapon_toggles` to work across all complevels as well. The idea is to eliminate this inconsistency unless a strict vanilla demo is being recorded.

For reference, the KEX port, even with the vanilla IWADs, allows all weapons to be cycled regardless of direction. It pops up a weapon carousel during weapon cycling, so it makes logical sense to do it that way.

Reply:
It could be a weapon menu option (I'm going to reorganize this for gamepad weapon groups anyway), just a rough idea:

`doom_weapon_cycling`
0 = Vanilla
1 = Full (default)
Disabled/Grayed out when vanilla strict mode is enabled, just like `doom_weapon_toggles` is grayed out.

Reply:
> So my question is, do we HAVE to follow Chocolate Doom's method for complevel 2? Going from Pistol to Shotgun directly (skipping SSG) is an unfair advantage, right?

Absolutely not, we can come up with any scheme. I think we have already improved the controls a lot in terms of mouse and gamepad support, and we should continue to do so.

Reply:
I think we must absolutely respect the fact that directly switching to the SG was impossible in Vanilla if the SSG was available. This is one of the main differences in game play introduced by Boom and there was a whole new code path introduced to make this possible.

Reply:
> I think we must absolutely respect the fact that directly switching to the SG was impossible in Vanilla if the SSG was available.

I suggest not to put such restrictions on the gamepad and the "next/prev weapon" buttons, they do not exist in Vanilla.

Reply:
I'll revisit this topic later. I still think there's room for improvement here without causing any issues with vanilla behavior.